polaris-react/src/components/Text/Text.stories.tsx

```diff
@@ -105,6 +105,15 @@ export const WithColor = () => (
   </LegacyStack>
 );
 
+export const WithInheritance = () => (
+  <Text as="p" variant="heading2xl" color="warning">
```

Comment:
`<p>` tags can't have nested block elements, so I think we'll need to update this to something like:

```tsx
<Text as="span" variant="heading2xl">
  <Text as="p">This is a 2xl heading</Text>
  <Text as="p">This is also a 2xl heading</Text>
</Text>
```

Reply:
Can they have flex elements? It's not a block level element per se. Also I didn't think flow content could be nested in phrasing content elements. So many rules

Reply:
Oh because it renders as a div and doesn't care about css, right? I just checked w3c though and it's the same thing for `span` > `p` 😞 

![image](https://screenshot.click/17-49-0a7ga-6euug.png)